Uchu.World/Objects/Components/ReplicaComponents/InventoryComponent.cs

```diff
@@ -34,366 +51,267 @@ protected InventoryComponent()
                 OnUnEquipped.Clear();
             });
             
-            Listen(OnStart, () =>
+            Listen(OnStart, async () =>
+            {
+                await LoadAsync();
+            });
+        }
+
+        /// <summary>
+        /// Loads the inventory for the game object, only for non-player GameObjects does this automatically equip the
+        /// items as the component cannot know when the inventory manager component will be available to load the
+        /// items from.
+        /// </summary>
+        private async Task LoadAsync()
+        {
+            if (!(GameObject is Player))
             {
-                if (GameObject is Player) return;
-                
                 var component = ClientCache.GetTable<ComponentsRegistry>().FirstOrDefault(c =>
                     c.Id == GameObject.Lot && c.Componenttype == (int) ComponentId.InventoryComponent);
+                var itemTemplates = ClientCache.GetTable<Core.Client.InventoryComponent>()
+                    .Where(i => i.Id == component.Componentid && i.Itemid != default).ToArray();
 
-                var items = ClientCache.GetTable<Core.Client.InventoryComponent>().Where(i => i.Id == component.Componentid).ToArray();
-
-                foreach (var item in items)
+                foreach (var itemTemplate in itemTemplates)
                 {
-                    if (item.Itemid == default) continue;
+                    if (itemTemplate.Itemid == default)
+                        continue;
                     
-                    var lot = (Lot) item.Itemid;
-
-                    var componentId = lot.GetComponentId(ComponentId.ItemComponent);
+                    var lot = (Lot) itemTemplate.Itemid;
+                    var itemComponent = (await ClientCache.GetTableAsync<ItemComponent>()).First(
+                        i => i.Id == lot.GetComponentId(ComponentId.ItemComponent));
```

Comment:
`Lot.GetComponentId` is an expensive method to call because it results in a CDClient database call, and doing this for each call in the Linq statement is *very* slow. The previous code called `var componentId = lot.GetComponentId(ComponentId.ItemComponent);` once per Linq statement and then compared `i.Id == componentId`. Currently, loading worlds is very slow.

Reply:
Change is up and looks good.